src/state/routing/useRoutingAPITrade.ts

```diff
@@ -45,17 +47,34 @@ export function useRoutingAPITrade<TTradeType extends TradeType>(
     routerPreference,
   })
 
+  const shouldUseRoutingApiV2 = useRoutingAPIV2Enabled()
+
   const {
-    isError,
-    data: tradeResult,
-    currentData: currentTradeResult,
-  } = useGetQuoteQuery(queryArgs ?? skipToken, {
+    isError: isLegacyAPIError,
+    data: legacyAPITradeResult,
+    currentData: currentLegacyAPITradeResult,
+  } = useGetQuoteQuery(!shouldUseRoutingApiV2 ? queryArgs ?? skipToken : skipToken, {
```

Comment:
this one liner is a little hard for me - if you're not using v2 api, query args, unless they don't exist in which case skipToken. if you're using v2, skipToken. i think maybe `shouldUseRoutingApiV2 ? skipToken : queryArgs ?? skipToken` is more readable to me, though i also would be opposed to moving this to a seprate statement